Documentation/ExtensionArchitecture/Extbase/Reference/FileUpload.rst

```diff
@@ -0,0 +1,489 @@
+..  include:: /Includes.rst.txt
+
+..  index:: Extbase, FileUpload
+
+..  _extbase_fileupload:
+
+===========
+File upload
+===========
+
+Implementing file uploads / attachments to Extbase Domain Models
+has always been a bit of a challenge.
+
+While it is easy to access an existing file reference in a Domain Model,
+writing new files to the :ref:`FAL (File Access Layer) <t3coreapi:using-fal>`
+takes more effort.
+
+..  _extbase_fileupload_accessing:
+
+Accessing a file reference in an Extbase Domain Model
+-----------------------------------------------------
+
+You need two components for the structural information: The Domain
+Model definition and the TCA entry.
+
+..  literalinclude:: _FileUpload/_Blog.php
+    :caption: EXT:my_extension/Classes/Domain/Model/Blog.php
+
+..  literalinclude:: _FileUpload/_tx_myextension_domain_model_blog.php
+    :caption: EXT:my_extension/Configuration/TCA/tx_myextension_domain_model_blog.php
+
+Once this is set up, you can create/edit records through the TYPO3
+backend, attach a single or multiple files in it. Then using a normal
```

Comment:
Then maybe add "in the backend through Web > Page or Web > List" to clarify?

Reply:
Done (I left out "Web > Page" because you usually only have file-upload enriched Extbase models connected to records and not content elements I'd guess?